velox/common/compression/Compression.h

```diff
@@ -45,6 +48,201 @@ CompressionKind stringToCompressionKind(const std::string& kind);
 
 constexpr uint64_t DEFAULT_COMPRESSION_BLOCK_SIZE = 256 * 1024;
 
+static constexpr int32_t kUseDefaultCompressionLevel =
+    std::numeric_limits<int32_t>::min();
+
+class StreamingCompressor {
+ public:
+  virtual ~StreamingCompressor() = default;
+
+  struct CompressResult {
+    uint64_t bytesRead;
+    uint64_t bytesWritten;
+    bool outputTooSmall;
```

Comment:
Please add the description for outputTooSmall here. The function `compress` should describe the function behavior.